docs/exts/sphinx_script_update.py

```diff
@@ -0,0 +1,113 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+import hashlib
+import json
+import os
+import sys
+import tempfile
+from distutils.file_util import copy_file
+from typing import Dict
+
+import requests
+from sphinx.builders import html as builders
+from sphinx.util import logging
+
+log = logging.getLogger(__name__)
+
+
+def _gethash(string: str):
+    hash_object = hashlib.sha256(string.encode())
+    return hash_object.hexdigest()
+
+
+def _user_cache_dir(appname=None):
+    """
+    Return full path to the user-specific cache dir for this application
+    """
+    if sys.platform == "win32":
+        # Windows has a complex procedure to download the App Dir directory because this directory can be
+        # changed in window registry, so i use temporary directory for cache
+        path = os.path.join(tempfile.gettempdir(), appname)
+    elif sys.platform == 'darwin':
+        path = os.path.expanduser('~/Library/Caches')
+    else:
+        path = os.getenv('XDG_CACHE_HOME', os.path.expanduser('~/.cache'))
+    path = os.path.join(path, appname)
+    return path
+
+
+def fetch_and_cache(script_url: str, output_filename: str):
+    """Fetch URL to local cache and returns path."""
+    cache_key = _gethash(script_url)
+    cache_dir = _user_cache_dir("redoc-doc")
+    cache_metadata_filepath = os.path.join(cache_dir, "cache-metadata.json")
+    cache_filepath = os.path.join(cache_dir, F"{cache_key}-{output_filename}")
```

Comment:
```suggestion
    cache_filepath = os.path.join(cache_dir, f"{cache_key}-{output_filename}")
```

Doesn't matter but just to be consistent :)